File: Python/350__Intersection_of_Two_Arrays_II.py
class Solution(object):
    def intersect(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        dic = {}
        for i in nums1:
            if i in dic:
                dic[i] += 1
            else:
                dic[i] = 1
        result = []
        for i in nums2:
            if i in dic and dic[i] != 0:
                result.append(i)
                dic[i] -= 1
        return result
        

File: Java/350__Intersection_of_Two_Arrays_II.java
class Solution {
    public int[] intersect(int[] nums1, int[] nums2) {
        List<Integer> lst = new ArrayList<>();
        Arrays.sort(nums1);
        Arrays.sort(nums2);
        int i = 0, j = 0;
        while (i < nums1.length && j < nums2.length) {
            if (nums1[i] == nums2[j]) {
                lst.add(nums1[i]);
                i++;
                j++;
            } else if (nums1[i] > nums2[j]) {
                j++;
            } else {
                i++;
            }
        }
        int[] result = new int[lst.size()];
        i = 0;
        for (int val : lst) {
            result[i] = val;
            i++;
        }
        return result;
    }
}
